e2e/tests/admin/analytics/utm-tracking.test.ts

```diff
@@ -2,88 +2,135 @@ import {AnalyticsWebTrafficPage} from '@/admin-pages';
+            await expect(page.getByRole('option', {name: 'UTM Medium', exact: true})).toBeVisible();
+            await expect(page.getByRole('option', {name: 'UTM Campaign', exact: true})).toBeVisible();
+            await expect(page.getByRole('option', {name: 'Source', exact: true})).toBeVisible();
+        });
+
+        test('selecting filter field shows value options with visit counts', async ({page, browser, baseURL}) => {
+            await withIsolatedPage(browser, {baseURL}, async ({page: publicPage}) => {
+                const homePage = new HomePage(publicPage);
+                await homePage.goto();
+            });
+
+            const analyticsWebTrafficPage = new AnalyticsWebTrafficPage(page);
+            await analyticsWebTrafficPage.goto();
+            await analyticsWebTrafficPage.openFilterPopover();
+            await page.getByRole('option', {name: 'Source', exact: true}).click();
+
+            const directOption = page.getByRole('option', {name: 'Direct'});
+            await expect(directOption).toBeVisible({timeout: 15000});
+            await expect(directOption).toContainText(/\d+/);
+        });
+
+        test('click on source row adds source filter', async ({page, browser, baseURL}) => {
+            await withIsolatedPage(browser, {baseURL}, async ({page: publicPage}) => {
+                const homePage = new HomePage(publicPage);
+                await homePage.goto();
+            });
+
+            const analyticsWebTrafficPage = new AnalyticsWebTrafficPage(page);
+            await analyticsWebTrafficPage.goto();
+            await expect(analyticsWebTrafficPage.topSourcesCard).toContainText('Direct', {timeout: 15000});
+
+            await analyticsWebTrafficPage.clickSourceToFilter('direct');
+
+            await expect(analyticsWebTrafficPage.getActiveFilter('Source')).toBeVisible();
+        });
+
+        test('filter persists in url', async ({page, browser, baseURL}) => {
             await withIsolatedPage(browser, {baseURL}, async ({page: publicPage}) => {
                 const homePage = new HomePage(publicPage);
-                await homePage.gotoWithQueryParams({
-                    utm_source: 'newsletter',
-                    utm_medium: 'email',
-                    utm_campaign: 'launch'
-                });
+                await homePage.goto();
             });
 
             const analyticsWebTrafficPage = new AnalyticsWebTrafficPage(page);
             await analyticsWebTrafficPage.goto();
-            await analyticsWebTrafficPage.selectCampaignType('UTM sources');
+            await expect(analyticsWebTrafficPage.topSourcesCard).toContainText('Direct', {timeout: 15000});
 
-            await expect(analyticsWebTrafficPage.topSourcesCard).toContainText('newsletter');
+            await analyticsWebTrafficPage.clickSourceToFilter('direct');
+
+            const url = page.url();
+            expect(url).toContain('source=');
```

Comment:
wondering if this could look something like:

```
const url = new URL(page.url());
expect(url.searchParams.get('source')).toBe('direct');

// or 
expect(url.searchParams.has('source')).toBe(true);
```